wagtail/users/views/groups.py

```diff
@@ -166,6 +166,8 @@ class GroupViewSet(ModelViewSet):
     edit_view_class = EditView
     delete_view_class = DeleteView
 
+    template_prefix = "wagtailusers/groups/"
```

Comment:
The template names produced by the logic in `get_templates()` aren't always used and the actual template used would sometimes just be the `template_name` on the view class fallback due to inconsistent naming, e.g. the index results view for groups uses `results.html` instead of `index_results.html`. I could rename the templates to make it consistent, but I'm not sure it's worth the potential breaking change.

Reply:
I'd be in favour of updating these template names to match the conventions set by the base ModelViewSet.

Re the breaking change - I assume the concern is with people overriding these templates by path (i.e. making their own customised copy of `wagtailusers/groups/results.html` within their own app)? If so, I think we've always been clear that we don't formally support that, and people do it at their own risk (and should expect it to break between releases). We should mention it in the release notes, but it doesn't need any formal process beyond that.

(also, just as a sanity check - the mention of `listing_results.html` below should be `index_results.html`, right?)

Reply:
Oops, just saw this after I pushed the latest changes.

> I assume the concern is with people overriding these templates by path (i.e. making their own customised copy of wagtailusers/groups/results.html within their own app)?

Yep, or extending that template.

I agree I think it's fair to rename them, let me just do it in a moment.

Reply:
OK, renamed in 849dd5124b9aecf3b13e00fb16d8b0df382f811b.

That made it possible to rely on `template_prefix` override and leaving the view's `template_name` unmodified from the generic views:
71aa1d2a3ecb9753ea54e452a798dc6dd858a57f

That last commit is not really necessary though, but I think it's nice knowing that you can just override `template_prefix` and write your templates. This would also break any subclasses of those views if they are used outside of the viewsets, but I don't see a reason why we need to support it. Happy to drop that commit if desired.

Reply:
I'm in favour of keeping that last commit! I know I've previously advocated for keeping the basic generic views fully functional in their standalone form, rather than reliant on being part of a viewset - but when it comes to the stuff further up the chain that exists primarily for Wagtail's own functionality rather than a building block for end users, I think it's perfectly fine for those to live firmly in the world of viewsets. It would be a shame to have these neat patterns for writing concise expressive view code and not be able to take advantage of them :-)